File: QuasselDroidNG/src/main/java/de/kuschku/libquassel/primitives/serializers/BoolSerializer.java
package de.kuschku.libquassel.primitives.serializers;

import android.support.annotation.NonNull;

import java.io.IOException;
import java.nio.ByteBuffer;
import java.nio.channels.ByteChannel;

public class BoolSerializer implements PrimitiveSerializer<Boolean> {
    @NonNull
    private static final BoolSerializer serializer = new BoolSerializer();
    private BoolSerializer() {}
    @NonNull
    public static BoolSerializer get(){
        return serializer;
    }

    @Override
    public void serialize(@NonNull final ByteChannel channel, @NonNull final Boolean data) throws IOException {
        final ByteBuffer buffer = ByteBuffer.allocate(1);
        buffer.put(0, (byte) (data ? 0x01 : 0x00));
        channel.write(buffer);
    }

    @NonNull
    @Override
    public Boolean deserialize(@NonNull final ByteBuffer buffer) throws IOException {
        return buffer.get() > 0;
    }
}

File: lib/src/main/java/de/kuschku/libquassel/protocol/primitive/serializer/BoolSerializer.kt
package de.kuschku.libquassel.protocol.primitive.serializer

import de.kuschku.libquassel.quassel.QuasselFeatures
import de.kuschku.libquassel.util.nio.ChainedByteBuffer
import java.nio.ByteBuffer

object BoolSerializer : Serializer<Boolean> {
  override fun serialize(buffer: ChainedByteBuffer, data: Boolean, features: QuasselFeatures) =
    buffer.put((if (data) 0x01 else 0x00).toByte())

  override fun deserialize(buffer: ByteBuffer, features: QuasselFeatures) =
    buffer.get() != 0x00.toByte()
}
